### PR TITLE
Add Ruff, pre-commit, and a Makefile

### PR DESCRIPTION
**Pull request by glass-ships**

### CODE REVIEW

File: bumps/dream/diffev.py [hunk 1]
```diff
@@ -39,7 +39,13 @@ def njit(*args, **kw):
 EPS = 1e-6
 _DE, _SNOOKER, _DIRECT = 0, 1, 2
 
+<<<<<<< HEAD
```
Comment (bmaranville) [inline bumps/dream/diffev.py:42]: I'm surprised git let this get pushed... usually it stops me if a merge is not complete.

Comment (glass-ships) [reply bumps/dream/diffev.py]: yeah i've been having some weird behavior with git the last few days... not sure what's up

**Comment by glass-ships** [comment]
.... should `dream/Random123/doc` and stuff have made it into the repo here? 
that feels. like that should not have happened

File: bumps/dream/Random123/docs/cbrng.dox [hunk 2]
```diff
@@ -24,7 +24,7 @@ error or substitute an alternative compatible
 CBRNG.)
 
 The API descriptions below are generic, but apply to
-all the different @ref families "families" of 
+all the different @ref families "families" of
```
Comment (bmaranville) [inline bumps/dream/Random123/docs/cbrng.dox:27]: no formatting should be applied to any file in the Random123 folder.  It is a vendored dependency and not part of our source management.

**Comment by glass-ships** [comment]
Closing in favor of #193

**Pull request closed without merging**